trade-finance-manager-api/src/v1/controllers/party.controller.js

```diff
@@ -1,31 +1,33 @@
 const { generateTfmAuditDetails } = require('@ukef/dtfs2-common/src/helpers/change-stream/generate-audit-details');
+
 const api = require('../api');
-const { canDealBeSubmittedToACBS, submitACBSIfAllPartiesHaveUrn } = require('./deal.controller');
+const submitDealToACBS = require('../helpers/submit-deal-acbs');
 
 const updateParty = async (req, res) => {
   const { dealId } = req.params;
-  const partyUpdate = {
+  const dealUpdate = {
     tfm: {
       parties: req.body,
     },
   };
 
   try {
-    const updatedDeal = await api.updateDeal({ dealId, dealUpdate: partyUpdate, auditDetails: generateTfmAuditDetails(req.user._id) });
+    const auditDetails = generateTfmAuditDetails(req.user._id);
+    const tfmDeal = await api.updateDeal({ dealId, dealUpdate, auditDetails });
+
+    console.log('====5', tfmDeal);
 
-    if (updatedDeal.dealSnapshot) {
-      if (canDealBeSubmittedToACBS(updatedDeal.dealSnapshot.submissionType)) {
-        await submitACBSIfAllPartiesHaveUrn(dealId);
-      }
+    if (tfmDeal) {
+      await submitDealToACBS(tfmDeal);
```

Comment:
not your code again but return of `submitDealToACBS` should be a promise

Reply:
Have updated.